### PR TITLE
gopkg.in/yaml.v2 Denial of Service (DoS) [nerdctl is NOT affected by this CVE]

### PR DESCRIPTION
**Pull request by fahedouch**
gopkg.in/yaml.v2 has vulnerability https://nvd.nist.gov/vuln/detail/CVE-2019-11254.

need to upgrade gopkg.in/yaml.v2 to version 2.2.8 or higher.

Signed-off-by: Fahed DORGAA <fahed.dorgaa@gmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -170,6 +170,7 @@ require (
 	google.golang.org/grpc v1.47.0 // indirect
 	google.golang.org/protobuf v1.28.0 // indirect
 	gopkg.in/square/go-jose.v2 v2.5.1 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
```
Comment (AkihiroSuda) [inline go.mod:173]: compose-go needs to be fixed too
- https://github.com/compose-spec/compose-go/pull/153

**Comment by AkihiroSuda** [comment]
I don't think the CVE affects us. We don't launch a server that accepts YAML from an untrusted source.

**Comment by fahedouch** [comment]
> I don't think the CVE affects us. We don't launch a server that accepts YAML from an untrusted source.

Agree but it is always good to non-vulnerable version to avoid scanner noise. It looks like we should point to this fork  https://github.com/compose-spec/compose-go/pull/153/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R20 rather then upstream repo

**Comment by AkihiroSuda** [comment]
Needs rebase

File: go.mod (hunk 1 above)
Comment (fahedouch) [reply go.mod:177]: @AkihiroSuda  https://github.com/compose-spec/compose-go/pull/153 is still Draft, can we merge this before compose-go ?

File: go.mod [hunk 2]
```diff
@@ -47,12 +47,12 @@ require (
 	github.com/tidwall/gjson v1.14.1
 	github.com/vishvananda/netlink v1.2.1-beta.2
 	github.com/vishvananda/netns v0.0.0-20211101163701-50045581ed74
-	golang.org/x/crypto v0.0.0-20220525230936-793ad666bf5e
-	golang.org/x/net v0.0.0-20220615171555-694bf12d69de
-	golang.org/x/sync v0.0.0-20220601150217-0de741cfad7f
+	golang.org/x/crypto v0.0.0-20220513210258-46612604a0f9
+	golang.org/x/net v0.0.0-20220516133312-45b265872317
+	golang.org/x/sync v0.0.0-20220513210516-0976fa681c29
```
Comment (AkihiroSuda) [inline go.mod:52]: Why rollback?

File: go.mod (hunk 1 above)
Comment (AkihiroSuda) [reply go.mod:177]: Yes

File: go.mod (hunk 2 above)
Comment (fahedouch) [reply go.mod]: Sorry it was a rebase issue!

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**